lib/auth/server.ts

```diff
@@ -15,75 +15,63 @@ limitations under the License.
 */
 
 import * as bodyParser from 'body-parser';
+import { EventEmitter } from 'events';
 import * as express from 'express';
 import type { Socket } from 'net';
 import * as path from 'path';
 
 import * as utils from './utils';
 import { ExpectedError } from '../errors';
 
-const serverSockets: Socket[] = [];
-
-const createServer = ({ port }: { port: number }) => {
-	const app = express();
-	app.use(
-		bodyParser.urlencoded({
-			extended: true,
-		}),
-	);
-
-	app.set('view engine', 'ejs');
-	app.set('views', path.join(__dirname, 'pages'));
-
-	const server = app.listen(port);
-	server.on('connection', (socket) => serverSockets.push(socket));
-
-	return { app, server };
-};
-
-/**
- * By design (more like a bug, but they won't admit it), a Node.js `http.server`
- * instance prevents the process from exiting for up to 2 minutes (by default) if a
- * client keeps a HTTP connection open, and regardless of whether `server.close()`
- * was called: the `server.close(callback)` callback takes just as long to be called.
- * Setting `server.timeout` to some value like 3 seconds works, but then the CLI
- * process hangs for "only" 3 seconds (not good enough). Reducing the timeout to 1
- * second may cause authentication failure if the laptop or CI server are slow for
- * any reason. The only reliable way around it seems to be to explicitly unref the
- * sockets, so the event loop stops waiting for it. See:
- * https://github.com/nodejs/node/issues/2642
- * https://github.com/nodejs/node-v0.x-archive/issues/9066
- */
-export function shutdownServer() {
-	serverSockets.forEach((s) => s.unref());
-	serverSockets.splice(0);
-}
+export class LoginServer extends EventEmitter {
```

Comment:
Previously there wasn't a class, just the functions `createServer()` and `awaitForToken()` in `server.ts`, where `awaitForToken` would call `createServer`.

`createServer` expected to be told what port number to use, which was hardcoded elsewhere as 8989. Code elsewhere also printed a URL to the console (in case of web auth), which I think is nice (allows me to copy and paste to the right browser window, when I have multiple browsers/profiles logged in to different balena user accounts) and which was actually a user request in a past CLI issue:
```
$ ./bin/balena login
...
Logging in to balena-cloud.com
? How would you like to login? Web authorization (recommended)
Opening web browser for URL:
https://dashboard.balena-cloud.com/login/cli/http%253A%252F%252F127.0.0.1%253A56628%252Fauth
```

So `createServer` or `awaitForToken` had to return the port number that was allocated to the OS, but `awaitForToken` was an async function that awaited for the user to click a button on the web browser (and the URL containing the port number had to be printed before the web browser was opened). Anyway, a refactoring was needed and I that's how I eventually came up with LoginServer class as an EventEmitter.

The majority of the code is reorganised copy-and-paste.